src/MerchantCenter/PolicyComplianceCheck.php

```diff
@@ -0,0 +1,240 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\WC;
+use Automattic\WooCommerce\GoogleListingsAndAds\Google\GoogleHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\PluginHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\TargetAudience;
+
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class PolicyComplianceCheck
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter
+ *
+ * @since x.x.x
+ */
+class PolicyComplianceCheck implements Service {
+
+	use PluginHelper;
+
+	/**
+	 * The WC proxy object.
+	 *
+	 * @var wc
+	 */
+	protected $wc;
+
+	/**
+	 * @var GoogleHelper
+	 */
+	protected $google_helper;
+
+	/**
+	 * @var TargetAudience
+	 */
+	protected $target_audience;
+
+	/**
+	 * BaseController constructor.
+	 *
+	 * @param WC             $wc
+	 * @param GoogleHelper   $google_helper
+	 * @param TargetAudience $target_audience
+	 */
+	public function __construct( WC $wc, GoogleHelper $google_helper, TargetAudience $target_audience ) {
+		$this->wc              = $wc;
+		$this->google_helper   = $google_helper;
+		$this->target_audience = $target_audience;
+	}
+
+	/**
+	 * Check if the store website is accessed by all users for the controller.
+	 *
+	 * @return bool
+	 */
+	public function is_accessible(): bool {
+		$all_allowed_countries = $this->wc->get_allowed_countries();
+		$target_countries      = $this->target_audience->get_target_countries();
+
+		foreach ( $target_countries as $country ) {
+			if ( ! array_key_exists( $country, $all_allowed_countries ) ) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Check if the store sample product landing pages lead to a 404 error.
+	 *
+	 * @return bool
+	 */
+	public function has_page_not_found_error(): bool {
+		$headers = $this->get_landing_page_headers();
+		if ( ! empty( $headers ) && $headers[0] !== 'HTTP/1.1 200 OK' ) {
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Check if the store sample product landing pages has redirects through 3P domains.
+	 *
+	 * @return bool
+	 */
+	public function has_redirects(): bool {
+				$headers = $this->get_landing_page_headers();
+		if ( ! empty( $headesr ) && isset( $headers[0] ) ) {
+			if ( $headers[0] === 'HTTP/1.1 302 Found' ) {
+				// this is the URL where it's redirecting
+				return true;
+			}
+		}
+			return false;
+	}
+
+	/**
+	 * Get the headers of the store sample product landing pages.
+	 *
+	 * @return array
+	 */
+	public function get_landing_page_headers(): array {
+		$ids = wc_get_products(
+			[
+				'return' => 'ids',
+				'limit'  => 1,
+			]
+		);
+		if ( ! empty( $ids ) ) {
+			$response = wp_remote_get( $ids[0]->get_permalink() );
```

Comment:
When I tried to check the implementation of the frontend wp-data store after getting the `mc/policy_check` API response, it responded with 500 error after running `wp.data.select('wc/gla').getPolicyCheck()` via the Console on the browser DevTool. I guess there might be a problem here.

PHP error log:
```
CRITICAL Uncaught Error: Call to a member function get_permalink() on int in /var/www/html/wp-content/plugins/google-listings-and-ads/src/MerchantCenter/PolicyComplianceCheck.php:115
Stack trace:
#0 /var/www/html/wp-content/plugins/google-listings-and-ads/src/MerchantCenter/PolicyComplianceCheck.php(79): Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\PolicyComplianceCheck->get_landing_page_headers()
#1 /var/www/html/wp-content/plugins/google-listings-and-ads/src/API/Site/Controllers/MerchantCenter/PolicyComplianceCheckController.php(72): Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\PolicyComplianceCheck->has_page_not_found_error()
#2 /var/www/html/wp-includes/rest-api/class-wp-rest-server.php(1140): Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter\PolicyComplianceCheckController->Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter\{closure}()
#3 /var/www/html/wp-includes/rest-api/class-wp-rest-server.php(987): WP_REST_Server->respond_ in /var/www/html/wp-content/plugins/google-listings-and-ads/src/MerchantCenter/PolicyComplianceCheck.php on line 115
```

Reply:
As mentioned [here](https://github.com/woocommerce/google-listings-and-ads/pull/1611#discussion_r935744640) the return parameter needs to be removed, which defaults to returning an array of WC_Product instead of an array of ID's.

Reply:
Thanks for pointing this out. Updated.